javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,275 @@
+  }
+  return `You name is ${user.name}`;
+}
+
+function addActivity(activity) {
+  if (user.todo.includes(activity)) {
+    return 'You already have this activity in to-do';
+  } else {
+    user.todo.push(activity);
+    return `${activity} added to your to-do`;
+  }
+}
+
+function removeActivity(activity) {
+  const indexOfRemovingActivity = user.todo.indexOf(activity);
+  if (indexOfRemovingActivity >= 0) {
+    user.todo.splice(indexOfRemovingActivity, 1);
+    return `Removed ${activity} from your to-do`;
+  } else {
+    return 'There is no such activity in your to-do';
+  }
+}
+
+function listTodos() {
+  if (user.todo.length === 0) {
+    return 'Your todo is empty';
+  } else {
+    const todoItems = user.todo.join(' and ');
+    return `You have ${user.todo.length} todos:\n${todoItems}`;
+  }
+}
+
+function getCurrentDate() {
+  const today = new Date();
+  const options = {
+    day: 'numeric',
+    month: 'long',
+    year: 'numeric',
+  };
+  const formattedDate = today.toLocaleDateString('en-GB', options); //15 May 2024
+  const [day, month, year] = formattedDate.split(' ');
+  return `Today is ${day}. of ${month} ${year}`;
+}
+
+function calcMathExpression(mathExpression) {
+  if (isValidMathExpression(mathExpression)) {
+    return eval(mathExpression).toString();
+  } else {
+    return 'I am afraid I did not catch the numbers. Could you repeat please ';
+  }
+}
+
+function setTimer(timeInfo, command) {
+  //calculating milliseconds for a timer
+  const [timeValue, timeUnit] = timeInfo;
+  let timeInMilliseconds;
+  switch (timeUnit) {
+    case 'seconds':
+    case 'second':
+      timeInMilliseconds = timeValue * 1000;
+      break;
+    case 'minutes':
+    case 'minute':
+      timeInMilliseconds = timeValue * 60 * 1000;
+      break;
+    case 'hours':
+    case 'hour':
+      timeInMilliseconds = timeValue * 60 * 60 * 1000;
+      break;
+    default:
+      return 'Oh, I did not catch the time. Please use seconds, minutes, or hours.'; // Return for invalid time units
+  }
+
+  // Clear existing timer
+  if (user.activeTimerId !== null) {
+    clearTimeout(user.activeTimerId);
+  }
+  // Set the new timer
+  user.activeTimerId = setTimeout(() => {
+    console.log(getReply(command)); //for seeing the message in consol
+    //code for working artyom.js
+    // const response = getReply(command);
+    // artyom.say(response);
+    clearTimeout(user.activeTimerId);
+  }, timeInMilliseconds);
+
+  return `Timer set for ${timeValue} ${timeUnit}`;
+}
+
+function getNiceMessage() {
+  const nicePhrases = [
+    'You are an incredible person',
+    'You are a wonderful friend',
+    'Your positivity is infectious',
+    'You have a great sense of humor',
+    'You are a great coder',
+    'Regular expressions are a piece of cake for you',
+    'You will soon be working as a web developer',
+  ];
+  const randomIndex = Math.floor(Math.random() * nicePhrases.length);
+  return `${nicePhrases[randomIndex]}, ${user.name}`;
```

Comment:
If `user.name` is not set it will use "undefined", "null" etc. In my case it wrote it in the string as "no", or "now. You could run a check if name is defined and print it only if it is.
![Screenshot 2024-06-14 at 11 11 23 PM](https://github.com/Natata08/hyf-homework/assets/52400967/62a9bcbe-6aa2-4d4c-af12-5ee29cd6eef9)

Reply:
I fixed it!